### ISSUE TITLE
[TTIRToTTNN] Integer ttir.SortOp routed through ttnn.sort compares keys in bf16, returning wrong sort indices

### ISSUE THREAD

**Issue by sonalibaskaran2499** [tenstorrent/tt-metal#46331]
## Describe the bug

`ttnn.sort` returns a **correct sorted-values tensor but an incorrect `indices` tensor** whenever the sort keys are integers whose magnitude exceeds ~256. The bitonic sort reuses the TopK SFPU sort LLK, which compares keys in **bf16** (8-bit mantissa). Keys that are not exactly representable in bf16 (e.g. distinct integers `> 256`) collide into ties, and because the `stable` flag is ignored, the indices of the tied keys are ordered arbitrarily.

The result: `ttnn.sort(...).values` looks correct (swapping two equal-in-bf16 values keeps the row "sorted"), but `ttnn.sort(...).indices`i.e. the `argsort` permutation is wrong. The dtype-cast operand workaround (`createSortOpOperandsWorkarounds`, casts integer input → `ui16`) only makes the op *run*; it does not fix the comparison precision, so the indices are still wrong.

## Downstream impact

Breaks `qwen_2_5_vl/pytorch-3B_Instruct` inference in tt-xla. The vision tower undoes its window-attention reordering with `reverse = torch.argsort(window_index); merged = merged[reverse]`, where `window_index` is an `int64` permutation of `0..550`. The wrong `argsort` indices scatter every image-patch embedding to the wrong token position, giving full-model logits **PCC ≈ 0.24** (required ≥ 0.99). Any model that uses sort/argsort **indices** over integer or `>256`-valued keys is affected.

## Root cause

`sort_single_row_single_core.cpp` does the compare/swap with the TopK SFPU LLK (`ckernel::topk_local_sort` / `topk_merge`), which compares keys in **bf16**, so distinct integers `> 256` collide into ties and since the indices ride the same compare/swap network, those ties scramble the indices.

## Steps to reproduce

In tt-xla (the integer-argsort regression test):

```
git checkout sonalib/argsort
pytest -svv tests/torch/ops/test_sort.py
```

## Sanity Test

Minimal `ttnn`-level reproducer (no tt-xla / MLIR needed):

```python
import torch
import ttnn

device = ttnn.open_device(device_id=0)

def check(n, hi, torch_dtype):
    # distinct keys: a permutation of [0, n) spread into [0, hi)
    perm = torch.randperm(n)
    x = (perm.to(torch.float64) / max(n - 1, 1) * hi).to(torch_dtype).reshape(1, n)

    tt = ttnn.from_torch(x, device=device, layout=ttnn.TILE_LAYOUT)
    _, tt_idx = ttnn.sort(tt, dim=-1)
    dev_idx = ttnn.to_torch(tt_idx).to(torch.int64)

    golden = torch.argsort(x.to(torch.float64), dim=-1)
    match = (dev_idx == golden).float().mean().item()
    print(f"n={n:4d} hi={hi:4.0f} {str(torch_dtype):16s} index_match={match:.4f}")

check(256, 255, torch.float32)   # values < 256  -> expect ~1.0  
check(512, 511, torch.float32)   # values > 256 
check(551, 550, torch.int32)     # qwen window_index size    

ttnn.close_device(device)
```

**Comment by sdjordjevicTT** [comment]
Hi @mgajewskiTT, are you aware of this issue on the metal side? Is this something that will be fixed sometime soon?

**Comment by mgajewskiTT** [comment]
Hi @sdjordjevicTT, I think this is connected to: https://github.com/tenstorrent/tt-metal/issues/37571, I though that was resolved a while ago, if that's not the case please contact the person who closed the ticket. Our team currently moved to data center and exabox work, we would need to look for someone else to resolve this.

**Comment by skhatriTT** [comment]
Hey @fahmedchanTT can you please help triage this issue to the right owner? Thank you!
